### PR TITLE
module-1

### PR DESCRIPTION
**Pull request by StanislavTrush**

### CODE REVIEW

File: src/main/java/com/epam/jmp2017/constants/BaseConstants.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	//filenames
+	public static final String FILE_ACTIONS = "actions.json";
+
+	//action types
+	public static final String PRINT = "PRINT";
+	public static final String PRINT_UPPER = "PRINT_UPPER";
+
+	//operations
+	public static final String EQUALS = "EQUALS";
```
Comment (AliaksandrPimenauAP) [inline src/main/java/com/epam/jmp2017/constants/BaseConstants.java]: I think it's enough to move these literals to enum (it is used in enum only)

File: src/main/java/com/epam/jmp2017/model/Action.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        this.conditions = conditions;
+    }
+
+    private boolean check(Data data) {
+        Attribute attribute;
+        Operation operation;
+        String attributeRealValue;
+        String attributeExpectedValue;
+        if(data != null) {
+            for (Condition condition : conditions) {
```
Comment (AliaksandrPimenauAP) [inline src/main/java/com/epam/jmp2017/model/Action.java:46]: logic can be moved to Condition class

File: src/main/java/com/epam/jmp2017/util/Worker.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    public String performActions(String dataString) throws IOException {
+        List<Result> results = new ArrayList<>();
+        List<Data> dataList = new ArrayList<>();
+        ClassLoader classLoader = getClass().getClassLoader();
+        URL actionsUrl = classLoader.getResource(BaseConstants.FILE_ACTIONS);
+        Gson gson = new Gson();
+        JsonParser parser = new JsonParser();
+        JsonReader reader = new JsonReader(new StringReader(dataString));
+        reader.setLenient(true);
+        if(!dataString.isEmpty()) {
```
Comment (AliaksandrPimenauAP) [inline src/main/java/com/epam/jmp2017/util/Worker.java]: can be splitted to 2 methods: parse Data list and parsing actions

Comment (StanislavTrush) [reply src/main/java/com/epam/jmp2017/util/Worker.java]: done

File: src/main/java/com/epam/jmp2017/constants/BaseConstants.java (hunk 1 above)
Comment (StanislavTrush) [reply src/main/java/com/epam/jmp2017/constants/BaseConstants.java]: done

**Pull request merged**